app/models/concerns/background_deletion.rb

```diff
@@ -5,32 +5,17 @@ module BackgroundDeletion
 
   included do
     class_attribute :background_deletion, default: [], instance_writer: false
+    class_attribute :background_deletion_method, default: :destroy!, instance_writer: false
+    class_attribute :background_deletion_scope_name, default: :all, instance_writer: false
```

Comment:
This method is never actually called from somewhere AFAIK. Only `Alert` references it but from a comment. I'm fine if you want to keep it anyway, but I suggest to remove it in order to simplify the implementation. We can get it back in the future if we find an scenario where we need it.